### PR TITLE
addVersion should only add __v__ param to same-origin requests

### PR DESCRIPTION
**Pull request by markbrocato**
React Storefront adds a ?__v__ parameter to all fetch and XHR requests to ensure that stale API responses are not served from the browser cache. This parameter should not be added to request to 3rd parties, because some 3rd parties such as Firebase will not accept unknown query parameters.

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
 // from previous versions of the app aren't served to new versions.
 /* istanbul ignore else */
 if (typeof window !== 'undefined') {
   const originalFetch = window.fetch
 
-  window.fetch = function rsfVersionedFetch(url, init) {
-    if (url.url) {
-      // the first param can be a request object
-      url = new Request(addVersion(url.url).toString(), url)
+  window.fetch = function rsfVersionedFetch(url, init, ...others) {
```
Comment (dijs) [inline src/fetch.js:25]: Did you forget to remove the `...others`?

File: src/fetch.js [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+    const parsed = getURL(url)
+
+    if (!isSameOrigin(parsed)) {
```
Comment (dijs) [inline src/fetch.js:28]: Why do you need to check for same origin here if the `addVersion` is always going to check this? Seems like you are doing double the work.

File: test/plp/SearchResultsProvider.test.js [hunk 3]
```diff
@@ -174,7 +174,7 @@ describe('SearchResultsProvider', () => {
         await wrapper.update()
       })
 
-      expect(fetch).toHaveBeenCalledWith('/api/test?__v__=development')
+      expect(fetch).toHaveBeenCalledWith('/api/test')
```
Comment (dijs) [inline test/plp/SearchResultsProvider.test.js:177]: Why was this test changed?

File: src/fetch.js (hunk 2 above)
Comment (markbrocato) [reply src/fetch.js:28]: You're correct, I've removed the sameOrigin check from addVersion.

File: src/fetch.js (hunk 1 above)
Comment (markbrocato) [reply src/fetch.js]: Fixed

File: test/plp/SearchResultsProvider.test.js (hunk 3 above)
Comment (markbrocato) [reply test/plp/SearchResultsProvider.test.js]: It broke when adding the sameOrigin test to addVersion, but I can revert the test now that I've removed the sameOrigin test from addVersion.

**Pull request merged**